account_payment_partner/wizard/payment_order_create.py

```diff
@@ -30,8 +30,25 @@ class PaymentOrderCreate(models.TransientModel):
     def extend_payment_order_domain(self, payment_order, domain):
         res = super(PaymentOrderCreate, self).extend_payment_order_domain(
             payment_order, domain)
-        domain += ['|', '|',
-                   ('invoice', '=', False),
+        # Monkey patch for fixing problem with the core search function
+        # when args has ('invoice', '=', False), referred in the issue #4857
+        # (https://github.com/odoo/odoo/issues/4857)
+        #
+        # Original domain:
+        # domain += ['|', '|',
+        #            ('invoice', '=', False),
+        #            ('invoice.payment_mode_id', '=', False),
+        #            ('invoice.payment_mode_id', '=', payment_order.mode.id)]
+        self.env.cr.execute(
+            'SELECT l.id '
+            'FROM account_move_line l '
+            'LEFT OUTER JOIN account_invoice i '
+            'ON l.move_id = i.move_id '
+            'WHERE i.id IS NULL')
```

Comment:
The memory footprint is very short and only for the live of the query, so I beg you to reconsider your decission. There's no other way to make it, and remember that this breaks standard import functionality that works without this module. This should be only temporary also.